<a href="https://colab.research.google.com/github/Joan-Njoki-Mwangi/Blinkit-Grocery-Store-Deep-Dive/blob/main/Blinkit_Grocery_Store_Deep_Dive.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>



## **Blinkit Analysis Overview**

> The objective of this analysis is to investigate customer penetration, feedback trends, and the impact of marketing campaigns, while also examining store demand and supply dynamics using Blinkit Grocery Store data points. The study further aims to segment customers based on behavioral patterns and to develop forecasts for future sales performance.

> The analysis utilizes five distinct datasets, which are merged in multiple ways to reveal hidden patterns. Particular focus is placed on understanding delivery times and their relationship to customer feedback, as well as identifying correlations with churn rates over time.

> Tools and Technologies: Python for advanced analysis and modeling, and Tableau for visualization and insight communication


### *Importing libraries*

In [5]:
# importing libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats
import warnings
warnings.filterwarnings('ignore')
%matplotlib inline

### 1.Customer segmentation & behavior

In [6]:
customer_seg = pd.read_csv("/content/blinkit_customers.csv")

## *Data Exploration*

In [7]:
# previewing five random rows
customer_seg.sample(5)

customer_id  customer_name                       email         phone  \
1703     49915498  Harini Mangal        aahana49@example.com  910459948075   
1550     56195475   Falguni Date  sharmaaradhana@example.com  913326874282   
2390     57217288   Dalaja Bassi         ppandey@example.net  915745364286   
2046     43431724    Jack Kalita      calebdewan@example.com  912843679556   
1655     50650215    Isaac Samra         aashi06@example.net  914073322680   

                                            address              area  \
1703            H.No. 932\nBaral Road\nKatni 070915         Jalandhar   
1550          94\nVenkataraman Ganj, Patiala-174200         Ahmedabad   
2390  339\nRadhakrishnan Circle, Bhubaneswar 566268        Karimnagar   
2046            41/500, Sachdev Marg, Jhansi-452338            Rohtak   
1655             H.No. 68, Bajaj Road\nDehri 969139  Pimpri-Chinchwad   

      pincode registration_date customer_segment  total_orders  \
1703   107053        2023-05-19          Premium             2   
1550    31357        2024-06-25          Regular            18   
2390   202075        2023-03-24          Regular             3   
2046   217110        2023-11-26              New             2   
1655   550671        2023-11-16              New            12   

      avg_order_value  
1703           950.88  
1550          1149.91  
2390           736.05  
2046           813.00  
1655          1474.41

In [8]:
customer_seg.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2500 entries, 0 to 2499
Data columns (total 11 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   customer_id        2500 non-null   int64  
 1   customer_name      2500 non-null   object 
 2   email              2500 non-null   object 
 3   phone              2500 non-null   int64  
 4   address            2500 non-null   object 
 5   area               2500 non-null   object 
 6   pincode            2500 non-null   int64  
 7   registration_date  2500 non-null   object 
 8   customer_segment   2500 non-null   object 
 9   total_orders       2500 non-null   int64  
 10  avg_order_value    2500 non-null   float64
dtypes: float64(1), int64(4), object(6)
memory usage: 215.0+ KB


In [9]:
#converting registration_date into datetime format
customer_seg['registration_date'] = pd.to_datetime(customer_seg['registration_date'])


-The dataset contains no null values, the columns in the correct data types, and the column names clearly defined.

## *Key insights*

In [10]:
# descriptive statistics
customer_seg.describe()

customer_id         phone        pincode           registration_date  \
count  2.500000e+03  2.500000e+03    2500.000000                        2500   
mean   4.989961e+07  9.150532e+11  497653.183200  2024-01-15 18:06:20.160000   
min    3.181300e+04  9.100132e+11     666.000000         2023-03-16 00:00:00   
25%    2.402443e+07  9.125946e+11  257079.750000         2023-08-21 18:00:00   
50%    5.000044e+07  9.150236e+11  498662.000000         2024-01-22 12:00:00   
75%    7.595459e+07  9.176256e+11  740668.000000         2024-06-12 06:00:00   
max    9.989390e+07  9.199988e+11  999631.000000         2024-11-04 00:00:00   
std    2.919266e+07  2.882835e+09  281167.293249                         NaN   

       total_orders  avg_order_value  
count   2500.000000      2500.000000  
mean      10.491600      1102.375836  
min        1.000000       200.430000  
25%        6.000000       631.817500  
50%       10.000000      1118.650000  
75%       16.000000      1565.400000  
max       20.000000      1999.830000  
std        5.805533       523.043183

### *Registrations*

In [11]:
customer_seg.registration_date =customer_seg['registration_date'].dt.strftime('%Y-%m-%d')

In [12]:
date_counts = customer_seg['registration_date'].value_counts()

# Identify the most occurring date and its frequency

most_occurring_date = date_counts.idxmax()
frequency = date_counts.max()

for date, count in date_counts.items():
    if count == frequency:
        print(f"The peak registration date was {date}, with a total of {count} registrations.")



The peak registration date was 2024-04-22, with a total of 10 registrations.
The peak registration date was 2024-06-23, with a total of 10 registrations.
The peak registration date was 2024-03-10, with a total of 10 registrations.
The peak registration date was 2024-10-03, with a total of 10 registrations.
The peak registration date was 2024-03-18, with a total of 10 registrations.


> Registrations cover the period from March 16, 2023, to November 4, 2024, with the majority occurring in 2024. Notably, March 10, March 18, April 22, June 23, and October 10 each recorded 10 registrations.

> The highest total number of orders recorded is 20, while the majority of customers placed around 10 orders, with a variation of approximately ±5.8 standard deviations from the mean(approximately 10 orders). Notably, all registered customers placed at least one order.

> The average order value stands at 1,118.65, with a dispersion of about ± 523.04 around the mean, providing key insight into customer spending behavior

### *Area*

In [48]:
# creating a column of the highest spend using the average order value and total orders done
customer_seg['total_spent'] = customer_seg.total_orders * customer_seg.avg_order_value


In [54]:
# based on area
#area with the highest orders, could mean that it has the highest reach in terms of market penetration
# area with the highest order value

High_spending_areas = (
    customer_seg.groupby('area').agg(
        total_orders = ('total_orders','sum'),
        total_sales = ('total_spent','sum'),
        avg_spend = ('avg_order_value','mean'),
        num_customers = ('customer_id','nunique')
    ).sort_values(by='total_sales',ascending = False).head(10))

Least_spending_areas = (
    customer_seg.groupby('area').agg(
        total_orders = ('total_orders','sum'),
        total_sales = ('total_spent','sum'),
         avg_spend = ('avg_order_value','mean'),
        num_customers = ('customer_id','nunique')
    ).sort_values(by='total_sales',ascending = True).head(10))

print("Top 10 High Spending Areas:\n", High_spending_areas, "\n")
print("Bottom 10  Spending Areas:\n", Least_spending_areas)


Top 10 High Spending Areas:
              total_orders  total_sales    avg_spend  num_customers
area                                                              
Deoghar               200    232826.09  1139.368235             17
Orai                  190    229979.58  1116.716471             17
Chittoor              163    216974.06  1182.904615             13
Korba                 160    207739.78  1147.996923             13
Jalna                 175    203583.78  1273.023889             18
Burhanpur             159    201574.71  1289.944167             12
Bathinda              168    198542.49  1136.222941             17
Nandyal               162    188903.73  1167.437857             14
Bhagalpur             145    188006.04  1277.073636             11
Bulandshahr           150    180392.90  1269.042308             13 

Bottom 10  Spending Areas:
                 total_orders  total_sales    avg_spend  num_customers
area                                                               

**Insights from Area-Wise Spending (2023–2024)**

**High Performing Areas**

Deoghar recorded the highest total spend at 232,826, making it a stronghold in terms of revenue contribution.

Jalna stands out with the highest customer base (18 unique customers), each spending an average of 1,273.

> This suggests that Jalna offers both depth (volume of customers) and quality (above-average spend).

> It may be a strategic market for expansion/penetration given its strong customer engagement and spending levels above the 50th percentile of price dispersion.

**Emerging/Undervalued Areas**

Nashik, though listed among the bottom-10 areas by total sales, shows a notable average spend of 1,154 across just 4 customers.

This indicates that while the market is small in size, the existing customer base is relatively high-value.

> With proper outreach, customer acquisition in Nashik could unlock untapped revenue potential.

**Low Performing Areas (High Risk/Low Value)**

Areas like Mangalore, Muzaffarpur, and Srikakulam show very low average spend per customer (330–667) combined with small customer bases (1–4 customers).

> These locations may not currently justify aggressive investment, unless future campaigns indicate a potentially higher demand.

Strategic Takeaways

Deoghar = Revenue powerhouse (retain & strengthen).

Jalna = High-penetration, high-value — good for targeted promotions/loyalty programs.

Nashik = Small but promising — opportunity for customer acquisition.

Mangalore, Muzaffarpur, Srikakulam = Low ROI zones; require cost-benefit analysis before further investment.



### *Trends* over time:
*   Customer base; number of customers and registrations
*   Avg order value
*   Total orders
*   Total spending

